pkg/provisioner/ironic/ironic.go

```diff
@@ -580,6 +583,36 @@ func (p *ironicProvisioner) PreprovisioningImageFormats() ([]metal3v1alpha1.Imag
 	return formats, nil
 }
 
+func (p *ironicProvisioner) setExternalURL(driverInfo map[string]interface{}) map[string]interface{} {
```

Comment:
Let's start with making sure `driverInfo` does NOT already contain `external_http_url`? Which, I guess, implies we need to set it to `nil` explicitly?